tests/commands/envelope_test.py

```diff
@@ -355,3 +356,41 @@ def test_apply_from_user_and_email(self):
 
         self.assertTrue(env.sign)
         self.assertIs(env.sign_key, mock.sentinel.gpg_key)
+
+
+class TestSendCommand(unittest.TestCase):
+
+    """Tests for the SendCommand class."""
+
+    mail = textwrap.dedent("""\
+        From: foo@example.com
+        To: bar@example.com
+        Subject: FooBar
+
+        Foo Bar Baz
+        """)
+
+    def test_get_account_by_address_with_str(self):
+        cmd = envelope.SendCommand(mail=self.mail)
+        account = mock.Mock()
+        with mock.patch(
+                'alot.commands.envelope.settings.get_account_by_address',
+                mock.Mock(return_value=account)) as get_account_by_address:
+            cmd.apply(mock.Mock())
+        get_account_by_address.assert_called_once_with('foo@example.com',
```

Comment:
But the idea of `with` is to handle the closing of stuff and not a naming scope. Note that even `for` does the same:
```py
for i in range(100):
    pass
print(i)
```